src/plugins/intel_gpu/src/graph/condition.cpp

```diff
@@ -12,6 +12,48 @@
 namespace cldnn {
 GPU_DEFINE_PRIMITIVE_TYPE_ID(condition)
 
+static layout get_output_layout_from_inner_program(kernel_impl_params const& impl_param, size_t branch_idx) {
+    std::string branch_name = (branch_idx == 0) ? "true" : "false";
+    auto& outputs  = impl_param.inner_progs[branch_idx]->get_outputs();
+    auto& io_output_map  = impl_param.io_output_maps[branch_idx];
+
+    CLDNN_ERROR_NOT_EQUAL(impl_param.desc->id,
+                        "Count of branch_" + branch_name + " outputs",
+                        io_output_map.size(),
+                        "expected outputs size",
+                        1,
+                        "Branch_" + branch_name + " should have one output.");
+
+    auto inner_prim_id = io_output_map.at(0);
+    for (size_t idx = 0; idx < outputs.size(); idx++) {
+        if (outputs[idx]->id() == inner_prim_id) {
+            return outputs.at(idx)->get_output_layout();
+        }
+    }
+    OPENVINO_THROW("Not found output with prim_id: ", inner_prim_id);
+}
+
+static layout get_output_layout_from_inner_network(kernel_impl_params const& impl_param, size_t branch_idx) {
+    std::string branch_name = (branch_idx == 0) ? "true" : "false";
+    auto& outputs  = impl_param.inner_nets[branch_idx]->get_outputs();
+    auto& io_output_map  = impl_param.io_output_maps[branch_idx];
+
+    CLDNN_ERROR_NOT_EQUAL(impl_param.desc->id,
+                        "Count of branch_" + branch_name + " outputs",
+                        io_output_map.size(),
+                        "expected outputs size",
+                        1,
+                        "Branch_" + branch_name + " should have one output.");
+
+    auto inner_prim_id = io_output_map.at(0);
+    for (size_t idx = 0; idx < outputs.size(); idx++) {
+        if (outputs[idx]->id() == inner_prim_id) {
+            return outputs.at(idx)->get_output_layout();
+        }
+    }
+    OPENVINO_THROW("Not found output with prim_id: ", inner_prim_id);
+}
```

Comment:
Seems that we can merge the above two util functions.

Reply:
Refactoring the function. please review it again. thanks.